libraries/WDT/src/WDT.cpp

```diff
@@ -0,0 +1,72 @@
+#include "WDT.h"
+
+// Constructor
+APM3_WDT::APM3_WDT()
+{
+    // The watchdog only runs off of the LFRC
+    am_hal_clkgen_control(AM_HAL_CLKGEN_CONTROL_LFRC_START, 0);
+    am_hal_wdt_init(&watchdogConfig);
+}
+
+// Configure the watchdog timer
+void APM3_WDT::configure(uint8_t clock, uint8_t interrupt, uint8_t reset)
+{
+  WDT->CFG_b.CLKSEL = clock; 
+  WDT->CFG_b.INTVAL = interrupt; 
+  WDT->CFG_b.RESVAL = reset; 
+}
+
+// Set the LFRC clock frequency of the watchdog timer
+void APM3_WDT::setClock(uint8_t clock)
+{
+  WDT->CFG_b.CLKSEL = clock; 
+}
+
+// Set number of watchdog timer ticks allowed before a watchdog interrupt event is generated
+void APM3_WDT::setInterrupt(uint8_t interrupt)
+{
+  WDT->CFG_b.INTVAL = interrupt;   
+}
+
+// Set number of watchdog timer ticks allowed before the watchdog will issue a system reset
+void APM3_WDT::setReset(uint8_t reset)
+{
+  WDT->CFG_b.RESVAL = reset; 
+}
+
+// Enable the watchdog
+void APM3_WDT::start()
+{
+  // Enable the interrupt for the watchdog in the NVIC
+  NVIC_EnableIRQ(WDT_IRQn);
+  am_hal_wdt_start();
+}
```

Comment:
Consider adding this line (just in case some other code disabled them):
`am_hal_interrupt_master_enable();`

Reply:
Good idea. Added!